metaseq/modules/transformer_decoder_layer.py

```diff
@@ -199,6 +200,18 @@ def forward(
         Returns:
             encoded output of shape `(seq_len, batch, embed_dim)`
         """
+        if getattr(self.args, "sequence_parallel", False):
```

Comment:
To confirm I'm following...
* We are still calling into ModelParallelTransformerDecoderLayer's build_fc1, build_fc2, etc. methods.
* These components (fc1, fc2, etc.) are then passed into `SequeuceParallelTransformerBlock`
* Killing model parallel modules going forward would imply that all models will be trained with sequence parallel

Separately, have a nit about naming since we separately have `TransformerDecoderMultiLayerBlockModule` floating around - is that still needed? If not, can open a task to remove (assume fsdp wrap every layer?) and keep the current `SequeuceParallelTransformerBlock` naming.

Reply:
no we need to kill "TransformerDecoderMultiLayerBlockModule", I can send a separate PR for that